examples/ann/main.go

```diff
@@ -0,0 +1,238 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Bram Gruneir (bram+code@cockroachlabs.com)
+
+package main
+
+import (
+	"database/sql"
+	"flag"
+	"fmt"
+	"math"
+	"math/rand"
+	"time"
+
+	"github.com/cockroachdb/cockroach/security"
+	"github.com/cockroachdb/cockroach/security/securitytest"
+	"github.com/cockroachdb/cockroach/server"
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+var inputLayerNodes = flag.Int("input-layer-nodes", 32, "Number of nodes in the input layer.")
+var hiddenLayerNodes = flag.Int("hidden-layer-nodes", 9, "Number of nodes in the hidden layer.")
+var outputLayerNodes = flag.Int("output-layer-nodes", 32, "Number of nodes in the hidden layer.")
+var hiddenLayers = flag.Int("hidden-layer-count", 1, "Number of hidden layers.")
+var totalIterations = flag.Int("iterations", 100, "Number of iterations.")
+
+var rng *rand.Rand
+var maxInput int64
+var totalLayers int
+var db *sql.DB
+
+func getNodeID(layer, num int) int {
+	return (layer * 100000) + num
+}
+
+func getLayerNodeCount(layer int) int {
+	if layer == 0 {
+		return *inputLayerNodes
+	}
+	if layer == totalLayers-1 {
+		return *outputLayerNodes
+	}
+	return *hiddenLayerNodes
+}
+
+func addInput() int64 {
+	newInput := rng.Int63n(maxInput)
+	tempInput := newInput
+	var binary string
+	tx, err := db.Begin()
+	if err != nil {
+		log.Fatal(err)
+	}
+	for i := 0; i < *inputLayerNodes; i++ {
+		bit := tempInput & 1
+		tempInput = tempInput >> 1
+		if bit == 0 {
+			binary += "0"
+		} else {
+			binary += "1"
+		}
+		if _, err := tx.Exec(`UPDATE nodes SET value=$1 WHERE id=$2`, float64(bit), getNodeID(0, i)); err != nil {
+			log.Fatal(err)
+		}
+		// TODO(bram): Obviously, remove this once we have feed forward working.
```

Comment:
um, ok. it's not clear what the TODO is here. I suggest clarifying it before merging